web/src/Root.tsx

```diff
@@ -0,0 +1,672 @@
+        ? localStorage.getItem(INIT_SESSION_ID_STORAGE_KEY)
+        : "",
+      rootDidInitialWatchFetch: false,
+      selectedAppName: null,
+      snapshotInProgressApps: [],
+      themeState: {
+        navbarLogo: null,
+      },
+    }
+  );
+
+  /**
+   * Sets the Theme State for the whole application
+   * @param {Object} newThemeState - Object to set for new theme state
+   * @param {Function} callback - callback to run like in setState()'s callback
+   */
+  const setThemeState = (newThemeState?: ThemeState) => {
+    if (newThemeState) {
+      setState({
+        themeState: { ...newThemeState },
+      });
+    }
+  };
+
+  /**
+   * Gets the current theme state of the app
+   * @return {Object}
+   */
+  const getThemeState = () => {
+    return state.themeState;
+  };
+
+  /**
+   * Clears the current theme state to nothing
+   */
+  const clearThemeState = () => {
+    /**
+     * Reference object to a blank theme state
+     */
+    const EMPTY_THEME_STATE = {
+      navbarLogo: null,
+    };
+
+    setState({
+      themeState: { ...EMPTY_THEME_STATE },
+    });
+  };
+
+  const handleActiveInitSession = (initSessionId: string) => {
+    if (Utilities.localStorageEnabled()) {
+      localStorage.setItem(INIT_SESSION_ID_STORAGE_KEY, initSessionId);
+    }
+    setState({ initSessionId });
+  };
+
+  // TODO: delete if not used
+  // const handleActiveInitSessionCompleted = () => {
+  //   if (Utilities.localStorageEnabled()) {
+  //     localStorage.removeItem(INIT_SESSION_ID_STORAGE_KEY);
+  //   }
+  //   setState({ initSessionId: "" });
+  // };
+
+  const checkIsHelmManaged = async () => {
+    try {
+      const res = await fetch(`${process.env.API_ENDPOINT}/is-helm-managed`, {
+        headers: {
+          Authorization: Utilities.getToken(),
+          "Content-Type": "application/json",
+        },
+        method: "GET",
+      });
+      if (res.ok && res.status === 200) {
+        const response = await res.json();
+        setState({ isHelmManaged: response.isHelmManaged });
+      } else {
+        setState({ isHelmManaged: false });
+      }
+    } catch (err) {
+      console.log(err);
+      setState({ isHelmManaged: false });
+    }
+  };
+
+  const getPendingApp = async () => {
+    try {
+      const res = await fetch(`${process.env.API_ENDPOINT}/pendingapp`, {
+        headers: {
+          Authorization: Utilities.getToken(),
+          "Content-Type": "application/json",
+        },
+        method: "GET",
+      });
+      if (!res.ok) {
+        if (res.status === 401) {
+          Utilities.logoutUser();
+          return;
+        }
+        if (res.status === 404) {
+          return;
+        }
+
+        console.log(
+          "failed to get pending apps, unexpected status code",
+          res.status
+        );
+        return;
+      }
+      const response = await res.json();
+      const app = response.app;
+      // TODO: delete if not used
+      // setState({
+      //   pendingApp: app,
```

Comment:
this `pendingApp` thing doesn't seem to be used anywhere.

Reply:
for future: can checkIsHelmManage be a custom hook? I've seen it used in a few places

Reply:
yeah there's a hook for it now. going to move it out of the root component and put it into a router and have it check the same way auth will work.